src/modules/sharing/sharing.service.ts

```diff
@@ -1460,6 +1464,19 @@ export class SharingService {
     });
   }
 
+  async bulkRemoveSharings(
+    user: User,
+    itemIds: Sharing['itemId'][],
+    itemType: Sharing['itemType'],
+  ) {
+    await this.sharingRepository.bulkDeleteInvites(itemIds, itemType);
```

Comment:
How are you ensuring that `bulkDeleteInvites` is not allowing a user who is not the owner of an item to delete an invitation of that item?

Reply:
Hey @sg-gs, this is used in an internal method, so the functions that call it are checking that for us, but I get what you mean.

Unlike `sharings`, there is not a direct way to accomplish it with a simple DELETE, as we do not save the owner. If that item is already being shared (think of this as already having at least 1 row in `sharings` table), we can just use findOne to fetch 1 row and check the ownership before calling this.

However, 1 item can have rows in `sharing_invitations` without any in `sharings`. So we can add the file/folder relation to the sharingInvite model and make a DELETE with a join that checks correctly if user the owner of the folder or file. Also, we could fetch again the file/folder and check the ownership before deleting sharings, but we are already doing this on `moveFilesToTrash` and `moveFoldersToTrash`